From 8306be301a09e1054b451c6d2a5c3b5d2fe1b8da Mon Sep 17 00:00:00 2001
From: linmin <linmin@eswincomputing.com>
Date: Wed, 19 Jun 2024 14:21:34 +0800
Subject: [PATCH 094/109] WIN2030-15279:fix(i2c timeout,bmc 2.2):Bug fix for
 mpq8785 timeout

Changelogs:
1.Delete HAL_I2C_DeInit(&hi2c3) in som_reset_control(). Otherwise
  it will cause mpq8785 timeout on the linux side, because the I2C3 was
  DeInit on the MCU side, i.e, the pin was initialized as GPIO.
  So that the linux can not operate correctly on the i2c bus also.
  This scenario is produced when reboot cold.
2.Update the bmc version to 2.2

Change-Id: Idcb72e601ebacd097c212436bcd2b6cbb575effb
---
 Core/Inc/hf_common.h        | 2 +-
 Core/Src/hf_common.c        | 4 ++++
 Core/Src/hf_power_process.c | 1 -
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/Core/Inc/hf_common.h b/Core/Inc/hf_common.h
index 756eca7..e06897b 100644
--- a/Core/Inc/hf_common.h
+++ b/Core/Inc/hf_common.h
@@ -242,7 +242,7 @@ extern UART_HandleTypeDef huart3;
 #define MCU_MAC_IDX	2
 /* define ------------------------------------------------------------*/
 #define BMC_SOFTWARE_VERSION_MAJOR                   2
-#define BMC_SOFTWARE_VERSION_MINOR                   1
+#define BMC_SOFTWARE_VERSION_MINOR                   2
 
 #define MAGIC_NUMBER	0x45505EF1
 
diff --git a/Core/Src/hf_common.c b/Core/Src/hf_common.c
index 995124d..5f03158 100644
--- a/Core/Src/hf_common.c
+++ b/Core/Src/hf_common.c
@@ -1391,6 +1391,10 @@ uint32_t es_autoboot(void)
 			printf("pwr enable and last state is power on\n");
 			return 1;
 		}
+		else {
+			printf("pwr enable and last state is power off\n");
+
+		}
 	}
 	return 0;
 }
diff --git a/Core/Src/hf_power_process.c b/Core/Src/hf_power_process.c
index cb2c97b..5b6595b 100644
--- a/Core/Src/hf_power_process.c
+++ b/Core/Src/hf_power_process.c
@@ -230,7 +230,6 @@ void som_reset_control(uint8_t reset)
 		HAL_GPIO_Init(MCU_RESET_SOM_N_GPIO_Port, &GPIO_InitStruct);
 		uart_deinit(UART4);
 		uart_deinit(USART6);
-		HAL_I2C_DeInit(&hi2c3);
 		SPI2_MASTER_CS_LOW();
 		set_bootsel(0, 0);
 	} else {
-- 
2.25.1

